VirtoCommerce.CatalogModule.Data/Services/CatalogSearchServiceImpl.cs

```diff
@@ -180,7 +180,27 @@ protected virtual void SearchCatalogs(SearchCriteria criteria, SearchResult resu
                     }
                 });
 
-                result.Catalogs = catalogs.OrderBy(x => x.Name).ToList();
+                result.Catalogs = OrderItems(catalogs, criteria);
```

Comment:
use follow  code  for  sorting  instead switch case
` var sortInfos = criteria.SortInfos;
                if (sortInfos.IsNullOrEmpty())
                {
                    sortInfos = new[] { new SortInfo { SortColumn = "Name" } };
                }
                query = query.OrderBySortInfos(sortInfos);
`